rotors_gazebo_plugins/src/gazebo_wind_plugin.cpp

```diff
@@ -119,44 +135,156 @@ void GazeboWindPlugin::OnUpdate(const common::UpdateInfo& _info) {
+    // Check if aircraft is out of wind field or not, and act accordingly.
+    if (x_inf >= 0u && y_inf >= 0u && vertical_factors_max >= 0u && 
+        x_sup <= (n_x_ - 1u) && y_sup <= (n_y_ - 1u) && vertical_factors_min <= 1u) {
+      // Find indices in z-direction for each of the vertices. If link is not 
+      // within the range of one of the columns, set to lowest or highest two.
+      std::size_t idx_z[8] = {0u, static_cast<int>(vertical_spacing_factors_.size()) - 1u,
+                              0u, static_cast<int>(vertical_spacing_factors_.size()) - 1u,
+                              0u, static_cast<int>(vertical_spacing_factors_.size()) - 1u,
+                              0u, static_cast<int>(vertical_spacing_factors_.size()) - 1u};
+      for (std::size_t i = 0u; i < 4u; ++i) {
+        if (vertical_factors_columns[i] < 0u) {
+          // Link z-position below lowest grid point of that column.
+          idx_z[2u * i + 1u] = 1u;
+        } else if (vertical_factors_columns[i] >= 1u) {
+          // Link z-position above highest grid point of that column.
+          idx_z[2u * i] = vertical_spacing_factors_.size() - 2u;
+        } else {
+          // Link z-position between two grid points in that column.
+          for (std::size_t j = 0u; j < vertical_spacing_factors_.size() - 1u; ++j) {
+            if (vertical_spacing_factors_[j] <= vertical_factors_columns[i] && 
+                vertical_spacing_factors_[j + 1u] > vertical_factors_columns[i]) {
+              idx_z[2u * i] = j;
+              idx_z[2u * i + 1u] = j + 1u;
+              break;
+            }
+          }
+        }
+      }
+
+      // Extract the wind velocities corresponding to each vertex.
+      math::Vector3 wind_at_vertices[8];
+      for (std::size_t i = 0u; i < 8u; ++i) {
+        wind_at_vertices[i].x = u_[idx_x[i] + idx_y[i] * n_x_ + idx_z[i] * n_x_
+                                                                        * n_y_];
+        wind_at_vertices[i].y = v_[idx_x[i] + idx_y[i] * n_x_ + idx_z[i] * n_x_
+                                                                        * n_y_];
+        wind_at_vertices[i].z = w_[idx_x[i] + idx_y[i] * n_x_ + idx_z[i] * n_x_
+                                                                        * n_y_];
+      }
+
+      // Extract the relevant coordinate of every point needed for trilinear 
+      // interpolation (first z-direction, then x-direction, then y-direction).
+      float interpolation_points[14];
+      for (std::size_t i = 0u; i < 14u; ++i) {
+        if (i < 8u) {
+          interpolation_points[i] = (
+            top_z_[idx_x[i] + idx_y[i] * n_x_] - bottom_z_[idx_x[i] + idx_y[i] * n_x_])
+            * vertical_spacing_factors_[idx_z[i]] + bottom_z_[idx_x[i] + idx_y[i] * n_x_];
+        } else if (i >= 8u && i < 12u) {
+          interpolation_points[i] = min_x_ + res_x_ * idx_x[2u * (i - 8u)];
+        } else {
+          interpolation_points[i] = min_y_ + res_y_ * idx_y[i - 9u];
```

Comment:
Same for this part here, this is hard to read unless you know exactly what format you used, put your constants in constexpr.